b2

```diff
@@ -801,9 +806,9 @@ def upload_file(args):
             headers['X-Bz-Info-' + k] = b2_url_encode(v)
 
         try:
-            response = post_file(url, headers, local_file, exit_on_error=False, progress_bar=True)
+            response = post_file(url, headers, local_file, exit_on_error=False, progress_bar=(not quiet))
             print json.dumps(response, indent=4, sort_keys=True)
-            if 'fileId' in response:
+            if (not quiet) and ('fileId' in response):
```

Comment:
As you can tell, I tend to prefer more parens when there might be any question about operator precedence.  This case could go either way.